orangecontrib/prototypes/widgets/owneighbours.py

```diff
@@ -86,16 +86,16 @@ def apply(self):
             self.send("Neighbors", None)
             return
         distance = self.DISTANCES[self.distance_index]
-        new = distance(np.vstack((self.data, self.reference)))[:len(self.data),
-              len(self.data):]
-        l = list(np.argsort(new.flatten()))[::-1]
-        s = set()
-        while len(l) > 0 and len(s) < self.n_neighbors:
-            i = int(l.pop() / len(self.reference))
-            if self.data[i] not in self.reference or not self.exclude_reference:
-                s.add(i)
-        print(distance.name, s)
-        neighbours = self.data[list(s)]
+        dist = distance(np.vstack((self.data, self.reference)))[:len(self.data),
+               len(self.data):]
+        sorted_indices = list(np.argsort(dist.flatten()))[::-1]
```

Comment:
```py
sorted_indices = list(np.argsort(dist.ravel())[::-1])
```
It's much faster to reverse an array than a list. Also use ravel() when copy is not _required_.